### ISSUE TITLE
HTTPSClient::connected() hang when _sslClient is NULL

### ISSUE THREAD

**Issue by yaweiw** [arduino-libraries/AzureIoTHub#22]
uint8_t HTTPSClient::connected()
{
    return _sslClient->connected();
}
when _sslClient is null, HTTPSClient::connected() hangs indefinitely. This doesn't seem right. Can we have some notification via e.g. serial port rather than completely unresponsive?

**Comment by sandeepmistry** [comment]
Hi @yaweiw,

We can definitely add a NULL check for prevent it crashing. Could you please share an example sketch and steps to reproduce the issue. I'm curious as to how you are encountering this issue.

```c++
uint8_t HTTPSClient::connected()
{
  return (_sslClient ? _sslClient->connected() : false);
}
```

**Comment by yaweiw** [comment]
@sandeepmistry it can easily be reproduced by not having below lines in your ino file.
Adafruit_WINC1500SSLClient sslClient;
AzureIoTHubClient iotHubClient(sslClient);

**Comment by sandeepmistry** [comment]
I think a better fix is changing `HTTPAPI_Init` in [src/httpapi.cpp](https://github.com/arduino-libraries/AzureIoTHub/blob/master/src/httpapi.cpp#L16-L22) to fail the SSL client is NULL and logging an error.

Would you have time to submit a PR for this?

**Comment by yaweiw** [comment]
Ok. I'll have a look later.

**Comment by sandeepmistry** [comment]
Closed via #24.
